### ISSUE TITLE
Lazy loading with Checkbox selection not working if record count is less than 100 or more than 100

### ISSUE THREAD

**Issue by LokeshBoran** [hjalmers/angular-generic-table#251]
The last checkbox issue is fixed but now if let say my API returns 12 records and the page size is 10 then checkbox functionality is not working.
Please check [example](https://stackblitz.com/edit/angular-generic-table-lazyloading-with-checkbox-znmmyp?file=src/app/app.component.ts).
### Step to reproduce - 
Simply click on `select/deselect all checkbox` it won't work.

It only works if 
```
 recordsAll:  100,
          recordsAfterSearch: 100,
          recordsAfterFilter: 100,
          recordLength: 10
```

**Comment by LokeshBoran** [comment]
Yes, It is perfectly fine. As an end user let say I clicked on select all then it should select only visible records or action should be performed on visible records. Or it should be configurable. I think `lazyAllSelected` will be used for both cases.

**Comment by jigar-g** [comment]
Hey, @LokeshBoran I am Facing the same Issue. Did you have any Solution for this?

**Comment by LokeshBoran** [comment]
@jigar-g No, but I think @hjalmers fixing this in next build.

**Comment by hjalmers** [comment]
I've added a fix for this in v4.16.0 so please give it a try @LokeshBoran and @jigar-g and let me know if you have any issues or questions regarding the implementation. If the checkbox in the table header is checked, all rows that are fetched will be added to the selection once loaded. The "select all" checkbox will be unchecked when a row is unselected i.e. when not all rows are selected anymore. Even though the table will know which records have been selected client side, it's important to look at the `lazyAllSelected` property on the table if the server wants to know if the client thinks that all rows are selected or not.
